bochs/iodev/serial.cc

```diff
@@ -1538,8 +1539,10 @@ void bx_serial_c::rx_timer(void)
     if (BX_SER_THIS s[port].tty_id >= 0) FD_SET(BX_SER_THIS s[port].tty_id, &fds);
 #endif
   }
+
+  // N.B: Throttle inputs if the OS has not yet responded to us.
   if ((BX_SER_THIS s[port].line_status.rxdata_ready == 0) ||
-      (BX_SER_THIS s[port].fifo_cntl.enable)) {
+      ((BX_SER_THIS s[port].fifo_cntl.enable) && (BX_SER_THIS s[port].rx_fifo_end < 15))) {
```

Comment:
Not sure what your thoughts are on applying this to `BX_SER_MODE_RAW` - that's the only odd one out here.